### PR TITLE
developer guide updates

### PR DESCRIPTION
**Pull request by sheyman**
Updates for the developer guide content.

### CODE REVIEW

File: api-docs/concepts.rst [hunk 1]
```diff
@@ -13,9 +13,9 @@ Secrets
 ~~~~~~~~~~~~~~~~~~
 
 A secret can be any data that requires security conscious storage. This
-may be reflected as a key, credentials, config file, etc.
+may be reflected as a key, credential, config file, etc.
```
Comment (meker12) [inline api-docs/concepts.rst]: Replace "A secret.... may be .. etc."  with the following sentence: 

A secret can be any data that requires security conscious storage like a key, credential, or configuration file.

File: api-docs/concepts.rst [hunk 2]
```diff
@@ -26,53 +26,53 @@ Some examples of a secret may include:
   * SSH Keys
 
 The secret schema represents the actual secret or key that is presented
-to the application. Currently secrets can be in any format, but
+to the |product name| service. Currently secrets can be in any format, but
 additional functionality may become available in the future for known
```
Comment (meker12) [inline api-docs/concepts.rst]: Currently secrets can be in any format, but
 additional functionality may become available in the future for known
 types of symmetric or asymmetric keys, like SSL certificates.

Note --We try not to include future-facing statements in doc.  They often fail to get updated, and sometimes never come true. 

Secrets can be any format.   

Would be helpful to provide a few example formats.

File: api-docs/concepts.rst [hunk 3]
```diff
@@ -26,53 +26,53 @@ Some examples of a secret may include:
...
 types of symmetric or asymmetric keys, like SSL certificates.
 
-The
-following shows an example of a secret:
+The following shows an example of a secret:
```
Comment (meker12) [inline api-docs/concepts.rst]: The following example shows a secret that has been added to |product name| .

(Assuming that is what the example is showing ;-) )

File: api-docs/concepts.rst [hunk 4]
```diff
@@ -106,11 +92,11 @@ Container
 A container is a way to logically group secrets that may be of a similar
 type; for example, grouping a private key, certificate, and bundle for
 an SSL certificate in a single container.
-The containers resource is the organizational center piece of Barbican. It
+The containers resource is the organizational center piece of |product name|. It
 creates a logical object that can be used to hold secret references. This is helpful
-when having to deal with tracking and having access to hundreds of secrets.
+when having to deal with tracking large numbers of secrets.
```
Comment (meker12) [inline api-docs/concepts.rst]: The Container content seems kind of redundant.  Consider replacing with the following:

The containers resource is the organizational center piece of |product name| that simplifies secrets management in environments that have large numbers of secrets. 

A container is a logical object that can be used to store secret references that are related by relationship or type. 
For example you can create a single container to group a private key, certificate, and bundle for
 an SSL certificate. Containers simplify the task of managing large numbers of secrets resources.  

-

File: api-docs/concepts.rst [hunk 5]
```diff
@@ -118,7 +104,7 @@ Barbican supports 3 types of containers:
 Each of these types have explicit restrictions as to what type of secrets should be
 held within. These will be broken down in their respective sections.
 
-This guide will assume you will be using a local running development environment of Cloud keep_warnings.
+This guide will assume you will be using a local running development environment of |product name|.
```
Comment (meker12) [inline api-docs/concepts.rst]: This guide assumes that you are running |product name| in a local development environment.

File: api-docs/concepts.rst [hunk 6]
```diff
@@ -242,32 +228,22 @@ key pass phrases.
...
-that apply to their project. Cloud Keep can derive the project that a user belongs
+All users authenticated with |product name| are able to read the effective quota values
+that apply to their project. |product name| can derive the project that a user belongs
 to by reading the project scope from the authentication token.
 
 Service administrators can read, set, and delete quota configurations for each
-project known to Barbican.  The service administrator is recognized by his authenticated
-role.  The service administrator's role is defined in Barbican's policy.json file.
+project known to |product name|.  The service administrator is recognized by its authenticated
+role.  The service administrator's role is defined in |product name|'s policy.json file.
```
Comment (meker12) [inline api-docs/concepts.rst]: These permissions are available to an authenticated user that has the service administrator role. This role is defined in the |product name|  ``policy.json`` configuration file.

File: api-docs/concepts.rst [hunk 7]
```diff
@@ -242,32 +228,22 @@ key pass phrases.
...
 The default role for a service admin is "key-manager:service-admin".
 
-Quotas can be enforced for the following Cloud Keep resources: secrets, containers,
-orders, consumers, and CAs.  The configured quota value can be None (use the default),
+Quotas can be enforced for the following |product name| resources: secrets, containers,
+and consumers.  The configured quota value can be None (use the default),
```
Comment (meker12) [inline api-docs/concepts.rst]: None ==>  If no value has been specified, |product name| uses the default setting, and the quota value is set to *None* .

Then, it makes me wonder... what is the default value, how is it set. That might be answered elsewehere.

File: api-docs/general-api-info/authenticate.rst [hunk 8]
```diff
@@ -102,7 +101,7 @@ Instead, find the endpoint for your service and region in your service catalog a
 .. note::
    If you authenticated with username and password credentials, and the
    Identity service returns a 401 message requesting additional credentials,
-   your account is configured for multi-factor authentication.
+   your account could be configured for multi-factor authentication.
```
Comment (meker12) [inline api-docs/general-api-info/authenticate.rst]: your account "might be"

File: api-docs/general-api-info/role-based-access-control.rst [hunk 9]
```diff
@@ -66,12 +66,4 @@ observer
 
 audit
     Users with this role are only allowed access to the resource metadata.
```
Comment (meker12) [inline api-docs/general-api-info/role-based-access-control.rst]: Users with this role are allowed access only to the resource metadata.

Comment (sheyman) [inline api-docs/general-api-info/role-based-access-control.rst]: done

File: api-docs/concepts.rst [hunk 10]
```diff
@@ -26,53 +24,62 @@ Some examples of a secret may include:
...
++--------------+---------------------------------------------------------------+
 
 
 You can use one of the following methods to store a secret:
 
--  Submit a **POST** request against the secrets resource and include
-   the secret metadata in the ``payload`` parameter.
+-  Submit a **POST** request against the secrets resource. Include
+   the secret metadata in the JSON body and include the secret itself
+   in the ``payload`` parameter.
```
Comment (meker12) [inline api-docs/concepts.rst]: in the JSON body. Include the secret itself in the ....

**Pull request merged**